app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,82 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show edit update destroy]
+
+  # GET /reservations or /reservations.json
+  def index
+    # @reservations = Reservation.all
+    # render json: @reservations
+    @user_reservations = Reservation.where(user_id: params[:user_id])
+    @reservations_info = []
+    @user_reservations.each do |reservation|
+      @reservations_info.push({
+                                id: reservation.id,
+                                user: reservation.user.username,
+                                house: reservation.house.name,
+                                city: reservation.city.name,
+                                date: reservation.date
+                              })
+    end
+    render json: @reservations_info
+  end
+
+  # GET /reservations/1 or /reservations/1.json
+  def show; end
+
+  # GET /reservations/new
+  def new
+    @reservation = Reservation.new
+  end
+
+  # GET /reservations/1/edit
+  def edit; end
+
+  # POST /reservations or /reservations.json
+  def create
+    @reservation = Reservation.new(reservation_params)
+
+    respond_to do |format|
+      if @reservation.save
+        format.html { redirect_to reservation_url(@reservation), notice: 'Reservation was successfully created.' }
+        format.json { render :show, status: :created, location: @reservation }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @reservation.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /reservations/1 or /reservations/1.json
+  def update
+    respond_to do |format|
+      if @reservation.update(reservation_params)
+        format.html { redirect_to reservation_url(@reservation), notice: 'Reservation was successfully updated.' }
+        format.json { render :show, status: :ok, location: @reservation }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @reservation.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /reservations/1 or /reservations/1.json
+  def destroy
+    @reservation.destroy
+
+    respond_to do |format|
+      format.html { redirect_to reservations_url, notice: 'Reservation was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
- [ ] What happens if the reservation is not successfully destroyed?  Kindly, add some error handling in this action. 👍